snooty/util.py

```diff
@@ -539,12 +545,20 @@ def deserialize(
                         continue
 
                 filtered_fields[field.name] = deserialized_children
+            elif field.type == FileId and isinstance(node_value, str):
+                filtered_fields[field.name] = field.type(node_value)
             else:
                 # Ideally, we validate that the data types of the fields match the data types of the JSON node,
                 # but that requires a more verbose and time-consuming process. For now, we assume data types are correct.
                 filtered_fields[field.name] = node_value
 
-        return node_type((0,), **filtered_fields)
+        deserialized_node = node_type(placeholder_span, **filtered_fields)
+
+        # Finalize any needed node types and their fields
+        if isinstance(deserialized_node, n.Heading) and not deserialized_node.id:
+            deserialized_node.id = make_html5_id(deserialized_node.get_text().lower())
```

Comment:
I believe this also needs the spaces replaced by dashes!

Reply:
@mmeigs this is what `make_html5_id()` does, based on the `PAT_INVALID_ID_CHARACTERS` regex `[^\w_\.\-]`!

Reply:
oh whoops, thanks!